libs/common/src/test/java/org/opensearch/common/GlobTests.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common;
+
+import org.opensearch.test.OpenSearchTestCase;
+
+public class GlobTests extends OpenSearchTestCase {
+
+
+    public void testGlobMatchNoWildcard() {
+        assertTrue(Glob.globMatch("test", "test"));
+        assertFalse(Glob.globMatch("test", "testing"));
+    }
+
+
+    public void testGlobMatchWildcardAtBeginning() {
+        assertTrue(Glob.globMatch("*test", "thisisatest"));
+        assertFalse(Glob.globMatch("*test", "thisisatesting"));
+    }
+
+
+    public void testGlobMatchWildcardAtEnd() {
+        assertTrue(Glob.globMatch("test*", "testthisisa"));
+        assertFalse(Glob.globMatch("test*", "atestthisis"));
+    }
+
+
+    public void testGlobMatchWildcardAtMiddle() {
+        assertTrue(Glob.globMatch("test*ing", "testthisisaing"));
+        assertFalse(Glob.globMatch("test*ing", "testthisisa"));
+    }
+
+
+    public void testGlobMatchMultipleWildcards() {
+        assertTrue(Glob.globMatch("*test*", "thisisatesting"));
```

Comment:
Is something like te**st allowed and do we test this?

Reply:
According to the old code, this was allowed. 
![image](https://github.com/opensearch-project/OpenSearch/assets/22767753/3cd08bce-671a-474d-9f5c-7a25bea319bf)
 I have added corresponding tests.